peps/pep-0750.rst

```diff
@@ -414,48 +414,59 @@ except that all expressions are always implicitly wrapped with a ``lambda``:
     lexical scope, including local and global variables. Any valid Python expression
     can be used, including function and method calls.
 
-However, this PEP further proposes to extend the usual lexical scoping of a lambda-wrapped
-expression to that supported by
-`annotation scopes <https://docs.python.org/3/reference/executionmodel.html#annotation-scopes>`_,
-so as to minimize developer surprise. For the remainder of the section, *class scope* will be
-used to refer to annotation scope as it's used by tag strings; this naming will be discussed
-momentarily.
+However, this PEP further proposes to extend the usual lexical scoping of a
+lambda-wrapped expression in the interpolation, as seen with class definitions,
+to that supported by `annotation scopes
+<https://docs.python.org/3/reference/executionmodel.html#annotation-scopes>`_,
+so as to minimize developer surprise.
 
-Let's look at why this is important with a somewhat contrived example where a developer sets up
-how figures are rendered in a caption, using a ``html`` tag:
+Let's look at why using annotation scope is important with a somewhat contrived
+example where a developer sets up how figures are rendered in a caption, using a
+``html`` tag:
 
 .. code-block:: python
 
     class SomeConfig:
         tag = 'b'
         figure = html'<{tag}>Figure</{tag}>'
 
-If the scope is only the usual lexical scope for lambdas, when ``figure`` is finally evaluated,
-the name ``tag`` is not available. Contrast that scoping with what is seen in using a f-string:
+Unless the ``html`` function fully evaluates all interpolations before the class
```

Comment:
This is not true; it will fail even if executed as part of the class body.

```
>>> class X:
...     tag = "b"
...     l = lambda: tag
...     print(l())
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in X
  File "<stdin>", line 3, in <lambda>
NameError: name 'tag' is not defined
```

The rule is that scopes nested within class scopes cannot see the class scope at all (except if the nested scope is an annotation scope).

Reply:
Right, the equivalent with the below would be:
```
class SomeConfig:
    tag = 'b'
    figure = f'<{(lambda: tag)()}>Figure</{(lambda: tag)()}>'
```

which immediately fails with `NameError: name 'tag' is not defined`, because the scope of the lambda here is itself not annotation scope.

We can point this fact out.

Reply:
This section is now rewritten.